luaui/Widgets/unit_auto_group.lua

```diff
@@ -201,12 +213,44 @@ local function RemoveOneUnitFromGroupHandler(_, _, args)
 	return true
 end
 
+local function loadAutogroupPreset(_, _, args, data)
+	local pr = args[1]
+	if not presets[tonumber(pr)] then
+		return
+	end
+	local prevGroup = presets[currPreset]
+
+	currPreset = tonumber(pr)
+
+	Echo(Spring.I18N("ui.autogroups.presetSelected", {presetNum = currPreset}))
+	unit2group = presets[currPreset]
+
+	if not unit2group then
+		return
+	end
+
+	for _, uID in ipairs(Spring.GetTeamUnits(myTeam)) do
+		local unitDefID = GetUnitDefID(uID)
+		local gr = unit2group[unitDefID]
```

Comment:
same as "pr"

```suggestion
		local group = unit2group[unitDefID]
```
(and associated slight refactor)